docs/how-tos/rust_service.md

```diff
@@ -0,0 +1,334 @@
+# Rust Service
+
+## Preface
+
+This guide will explain how to develop a Rust service to serve data to the Graph.
+We will cover:
+
+- Setting up the HTTP/GraphQL Server in [Serving GraphQL Requests](#serving-graphql-requests)
+- Creating GraphQL Objects and providing field resolvers in [Defining Objects](#defining-objects)
+- Extending the graph using [Federation](#federated)
+- Exporting the GraphQL schema for use in federation in [Schema Generation](#schema-generation)
+
+## Dependencies
+
+This guide will utilize the following dependencies:
+
+- `axum` as the HTTP server
+- `async-graphql` as graphql server library 
+- `async-graphql-axum` for GraphQL request handling
+- `tokio` as runtime for writing asynchronous rust program
+
+```toml
+[dependencies]
+async-graphql = "7.0.6"
+async-graphql-axum = "7.0.6"
+axum = "0.7.5"
+tokio = { version = "1.38.0", features = ["macros", "rt-multi-thread"] }
+```
+
+## Serving GraphQL Requests
+
+To handle GraphQL requests, we start by creating a server with a placeholder GraphQL schema. We use the `async-graphql-axum` crate, which provides integration between `async-graphql` and `axum`, to create a `graphql_handler`.
+
+!!! example "GraphQL Request Handler"
+
+    ```rust
+    use async_graphql::{EmptyMutation, EmptySubscription, Object, Schema};
+    use async_graphql_axum::{GraphQLRequest, GraphQLResponse};
+    use axum::{
+        extract::Extension,
+        routing::post,
+        Router,
+    };
+
+    struct Query;
```

Comment:
Does this work without deriving anything on it?

Reply:
No, it's not a working code

Reply:
Ah, would be good to get it to the point where it works, as otherwise it will probably confuse people